### ISSUE TITLE
Android build error

### ISSUE THREAD

**Issue by MoNTE48** [minetest/minetest#3366]
Hi. Today, a new version of irrlicht ogl-es branch. Files in the patches that are made now removed. There is a bug in make.

**Comment by est31** [comment]
Can you paste the error here?

**Comment by MoNTE48** [comment]
Exemple, on new version remove this file irrlicht/source/Irrlicht/COGLESTexture.cpp

**Comment by est31** [comment]
I can confirm build failure. There have been recent commits to the ogl-es branch:

https://github.com/zaki/irrlicht/commits/ogl-es

This commit being pretty heavy (watch out, its large): https://github.com/zaki/irrlicht/commit/ce6b7562ce5dbcf742568cc2398bfb4410710311

Its not just our fault, but also theirs. For example, I see a missing define error for `video::EDT_DIRECT3D8` in `source/Irrlicht/Android/CIrrDeviceAndroid.cpp`, which seems to have been removed since from the `E_DRIVER_TYPE` enum definition in `deps/irrlicht/include/EDriverTypes.h`. So this is both our bug, and upstream bug.

I think the best we can do is wait a few days for the dust to settle, and if it still doesn't build, write a kind bug report to irrlicht. Then, once vanilla irrlicht works, we can look into updating the patches.

**Comment by MoNTE48** [comment]
I see new commits http://sourceforge.net/p/irrlicht/code/5185/log/?path=/branches/ogl-es
I think they fixed it. I can check it out tomorrow, when I'm in the office.
By the way, if I understand correctly, they broke support ogles1? This is a big problem, because about 5% of devices used ogles1, as well as many ogles 3 / 3.1 devices automatically use the driver ogles1. You may want to do irrlicht fork 5122 for compatibility?

**Comment by est31** [comment]
I don't think that it got fixed. At least I saw no change near the code I have seen the error earlier. Haven't tried though.

**Comment by MoNTE48** [comment]
https://github.com/minetest/minetest/commit/2f58e4d9624b4f31c5064dc640e0ad40d8adb9ea
Minetest crash on starting nativeactivity!

**Comment by est31** [comment]
It works for me. Can you make a backtrace?

**Comment by est31** [comment]
To update: revision 5198 (commit on git https://github.com/zaki/irrlicht/commit/58cba0b3c6de681ca35227e318223fc48af27aaa) still isn't working.

**Comment by MoNTE48** [comment]
Good news https://github.com/zaki/irrlicht/commit/878685affa9e017c03183ff4e5f34267d9f1ec48
(- Fixed version detecting mechanism for both OpenGL ES drivers.)

**Comment by MoNTE48** [comment]
jni/../jni/src/client/tile.cpp:939:43: note: candidates are:
In file included from deps/irrlicht/include/IrrlichtDevice.h:10:0,
                 from jni/../jni/src/client/tile.h:27,
                 from jni/../jni/src/client/tile.cpp:20:
deps/irrlicht/include/IVideoDriver.h:586:16: note: virtual bool irr::video::IVideoDriver::setRenderTarget(irr::video::IRenderTarget*, irr::u16, irr::video::SColor, irr::f32, irr::u8)
   virtual bool setRenderTarget(IRenderTarget* target, u16 clearFlag, SColor clearColor = SColor(255,0,0,0),
                ^
deps/irrlicht/include/IVideoDriver.h:618:16: note: virtual bool irr::video::IVideoDriver::setRenderTarget(irr::video::ITexture*, irr::u16, irr::video::SColor, irr::f32, irr::u8)
   virtual bool setRenderTarget(ITexture* texture, u16 clearFlag=ECBF_COLOR|ECBF_DEPTH, SColor clearColor = SColor(255,0,0,0),
                ^
deps/irrlicht/include/IVideoDriver.h:624:25: note: bool irr::video::IVideoDriver::setRenderTarget(irr::video::ITexture*, bool, bool, irr::video::SColor)
   _IRR_DEPRECATED_ bool setRenderTarget(ITexture* texture, bool clearBackBuffer, bool clearZBuffer, SColor color = SColor(255,0,0,0))
                         ^
In file included from deps/irrlicht/include/irrTypes.h:8:0,
                 from jni/src/irrlichttypes.h:34,
                 from jni/src/irrlichttypes_bloated.h:23,
                 from jni/src/network/connection.h:23,
                 from jni/src/client.h:23,
                 from jni/../jni/src/network/clientopcodes.h:24,
                 from jni/../jni/src/network/clientopcodes.cpp:21:
deps/irrlicht/include/IrrCompileConfig.h:99:0: warning: "_IRR_ANDROID_PLATFORM_" redefined [enabled by default]
 #define _IRR_ANDROID_PLATFORM_

**Comment by paramat** [comment]
@MoNTE48 Can this issue be closed now?

**Comment by MoNTE48** [comment]
@paramat,  As long as we are using an older version of irrlicht, I think not. There are still problems in the minetest on android with the latest irrlicht, but I did not have time posiotret what was going on. Perhaps it can do @sapier or @est31?
